rpcs3/Emu/Io/RB3MidiGuitar.cpp

```diff
@@ -0,0 +1,257 @@
+{
+	rtmidi_in_free(midi_in);
+}
+
+static u8 disabled_response[] = {
+	0xe9, 0x00, 0x00, 0x00, 0x00, 0x02, 0x0f, 0x01,
+	0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+	0x00, 0x00, 0x80, 0x00, 0x00, 0x00, 0x00, 0x82,
+	0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+	0x21, 0x26, 0x02, 0x06, 0x00, 0x00, 0x00, 0x00};
+
+static u8 enabled_response[] = {
+	0xe9, 0x00, 0x00, 0x00, 0x02, 0x00, 0x00, 0x00,
+	0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+	0x00, 0x00, 0x80, 0x00, 0x00, 0x00, 0x00, 0x8a,
+	0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+	0x21, 0x26, 0x02, 0x06, 0x00, 0x00, 0x00, 0x00};
+
+void usb_device_rb3_midi_guitar::control_transfer(u8 bmRequestType, u8 bRequest, u16 wValue, u16 wIndex, u16 wLength, u32 buf_size, u8* buf, UsbTransfer* transfer)
+{
+	transfer->fake = true;
+
+	// configuration packets sent by rock band 3
+	// we only really need to check 1 byte here to figure out if the game
+	// wants to enable midi data or disable it
+	if (bmRequestType == 0x21 && bRequest == 0x9 && wLength == 40)
+	{
+		if (buf_size < 3)
+		{
+			rb3_midi_guitar_log.warning("buffer size < 3, bailing out early");
+			return;
+		}
+
+		switch (buf[2])
+		{
+		case 0x89:
+			rb3_midi_guitar_log.notice("MIDI data enabled.");
+			buttons_enabled = true;
+			response_pos = 0;
+			break;
+		case 0x81:
+			rb3_midi_guitar_log.notice("MIDI data disabled.");
+			buttons_enabled = false;
+			response_pos = 0;
+			break;
+		default:
+			rb3_midi_guitar_log.warning("Unhandled SET_REPORT request: 0x%02X");
+			break;
+		}
+	}
+	// the game expects some sort of response to the configuration packet
+	else if (bmRequestType == 0xa1 && bRequest == 0x1)
+	{
+		transfer->expected_count = buf_size;
+		if (buttons_enabled)
+		{
+			const usz remaining_bytes = sizeof(enabled_response) - response_pos;
+			const usz copied_bytes = std::min<usz>(remaining_bytes, buf_size);
+			memcpy(buf, &enabled_response[response_pos], copied_bytes);
+			response_pos += copied_bytes;
+		}
+		else
+		{
+			const usz remaining_bytes = sizeof(disabled_response) - response_pos;
+			const usz copied_bytes = std::min<usz>(remaining_bytes, buf_size);
+			memcpy(buf, &disabled_response[response_pos], copied_bytes);
+			response_pos += copied_bytes;
+		}
+	}
+	else if (bmRequestType == 0x21 && bRequest == 0x9 && wLength == 8)
+	{
+		// the game uses this request to do things like set the LEDs
+		// we don't have any LEDs, so do nothing
+	}
+	else
+	{
+		usb_device_emulated::control_transfer(bmRequestType, bRequest, wValue, wIndex, wLength, buf_size, buf, transfer);
+	}
+}
+
+void usb_device_rb3_midi_guitar::interrupt_transfer(u32 buf_size, u8* buf, u32 /*endpoint*/, UsbTransfer* transfer)
+{
+	transfer->fake = true;
+	transfer->expected_count = buf_size;
+	transfer->expected_result = HC_CC_NOERR;
+	// the real device takes 8ms to send a response, but there is
+	// no reason we can't make it faster
+	transfer->expected_time = get_timestamp() + 1'000;
+
+	if (buf_size < 28)
```

Comment:
I don't get why you changed this to 28.
You clearly only need 27 everywhere below.